src/main/java/com/redhat/lightblue/camel/LightblueProducer.java

```diff
@@ -22,7 +24,20 @@ public LightblueProducer(LightblueScheduledPollEndpoint endpoint) {
     public void process(Exchange exchange) throws Exception {
         LightblueRequest req = (AbstractLightblueDataRequest) exchange.getIn().getBody();
 
-        exchange.getOut().setBody(endpoint.getLightblueClient().data(req));
+        try {
+            LightblueResponse response = endpoint.getLightblueClient().data(req);
+            if (response.hasError()) {
+                throw new RuntimeException("LightblueProducer:Error returned in response: " + response.getText());
```

Comment:
@dcrissman it is not a short lived exception, the RTE becomes a root cause of LightblueErrorResponseException. any who, I changed the code to set LightblueErrorResponseException on the exchange twice.